File: src/main/java/client/renderables/EntityRenderer.java
package client.renderables;

import client.Game;
import common.entities.Player;
import common.world.Direction;
import common.world.World;
import utils.FileUtils;

import javax.imageio.ImageIO;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.IOException;

public class EntityRenderer implements Entity, Drawable {
    public int offX, offY;
    private BufferedImage texture;
    private common.entities.Entity entity;
    private int x, width, y, height;
    private Direction prevDirection = Direction.LEFT;

    public EntityRenderer(common.entities.Entity entity, int x, int y) {
        try {
            this.texture = FileUtils.scale1(ImageIO.read(this.getClass().getClassLoader().getResource("tex/entities/" + entity.getName() + ".png")), 4.0);
            System.out.println(entity.getName());
        } catch (IOException | NullPointerException e) {
            try {
                this.texture = FileUtils.scale1(ImageIO.read(this.getClass().getClassLoader().getResource("tex/placeholder.png")), 4.0);
            } catch (IOException ex) {
                texture=new BufferedImage(64,64,BufferedImage.TYPE_INT_ARGB);
                ex.printStackTrace();
            }
            e.printStackTrace();
        }
        this.entity = entity;
        entity.setFacing(Direction.RIGHT);
        this.x = x;
        this.y = y;
        this.width = 64;
        this.height = 128;
        prevDirection = entity.getFacing();
    }

    @Override
    public void draw(Graphics g) {
        g.drawImage(texture, x, y-64,64,128, null);
    }

    @Override
    public void tick() {
    }

    @Override
    public void second() {
    }

    public common.entities.Entity getEntity() {
        return entity;
    }

    public int getY() {
        return y;
    }

    public int getX() {
        return x;
    }
}

File: src/main/java/client/renderables/EntityRenderer.kt
package client.renderables

import common.world.Direction
import utils.FileUtils

import javax.imageio.ImageIO
import java.awt.*
import java.awt.image.BufferedImage
import java.io.IOException

class EntityRenderer(val entity: common.entities.Entity, val x: Int, val y: Int) : RenderedEntity(), Drawable {
    var offX: Int = 0
    var offY: Int = 0
    private var texture: BufferedImage? = null
    private val width: Int
    private val height: Int
    private var prevDirection = Direction.LEFT

    init {
        try {
            this.texture = FileUtils.scale1(ImageIO.read(this.javaClass.classLoader.getResource("tex/entities/" + entity.name + ".png")!!), 4.0)
            println(entity.name)
        } catch (e: IOException) {
            try {
                this.texture = FileUtils.scale1(ImageIO.read(this.javaClass.classLoader.getResource("tex/placeholder.png")!!), 4.0)
            } catch (ex: IOException) {
                texture = BufferedImage(64, 64, BufferedImage.TYPE_INT_ARGB)
                ex.printStackTrace()
            }

            e.printStackTrace()
        } catch (e: NullPointerException) {
            try {
                this.texture = FileUtils.scale1(ImageIO.read(this.javaClass.classLoader.getResource("tex/placeholder.png")!!), 4.0)
            } catch (ex: IOException) {
                texture = BufferedImage(64, 64, BufferedImage.TYPE_INT_ARGB)
                ex.printStackTrace()
            }

            e.printStackTrace()
        }

        entity.facing = Direction.RIGHT
        this.width = 64
        this.height = 128
        prevDirection = entity.facing
    }

    override fun draw(g: Graphics) {
        g.drawImage(texture, x, y - 64, 64, 128, null)
    }

    override fun tick() {}

    override fun second() {}
}
